bin/cml/runner.js

```diff
@@ -559,7 +568,7 @@ exports.builder = (yargs) =>
       },
       destroyDelay: {
         type: 'number',
-        default: 20,
+        default: 10,
```

Comment:
what's the reason for this? Was 20 too much?

Reply:
ok but what's too little? Why not 5? Or 1? Or 0?

Reply:
> _`destroyDelay` waits a bit so the provider has time to [pull logs](https://github.com/iterative/terraform-provider-iterative/blob/af2082e9f95654145adf67db660c5fac2dca0877/iterative/resource_runner.go#L240-L266) a last time if the runner fails; rationale explained on [cml#419](https://github.com/iterative/cml/pull/419). It began being [30](https://github.com/iterative/cml/blob/ef5d9b9dfa952eedf2e1ead76cd18a09b5bc86be/bin/cml-runner.js#L21) seconds and then was changed to [20](https://github.com/iterative/cml/blob/96743a4df929da6fef32161f0c09643bde998b89/bin/cml-runner.js#L20)._

Reply:
https://github.com/iterative/cml/pull/689/files#r823121962

Reply:
On the provider side, see:
* [iterative/terraform-provider-iterative@`b47f801`/iterative/resource_runner.go:248](https://github.com/iterative/terraform-provider-iterative/blob/b47f8014c1a222362b43ac9c5afc8a9522fa7ea3/iterative/resource_runner.go#L248)
* [hashicorp/terraform-plugin-sdk@`9cecd71`/helper/resource/wait.go](https://github.com/hashicorp/terraform-plugin-sdk/blob/9cecd719095910f47ea6810329a18947dc6e353b/helper/resource/wait.go)
* [hashicorp/terraform-plugin-sdk@`9cecd71`/helper/resource/state.go](https://github.com/hashicorp/terraform-plugin-sdk/blob/9cecd719095910f47ea6810329a18947dc6e353b/helper/resource/state.go)
* [`resource.Retry` documentation](https://www.terraform.io/plugin/sdkv2/resources/retries-and-customizable-timeouts#retry)

Terraform waits a [maximum of 10 seconds](https://github.com/hashicorp/terraform-plugin-sdk/blob/9cecd719095910f47ea6810329a18947dc6e353b/helper/resource/state.go#L196) between checks.

Reply:
Still, SSH connection delays and other factors may decrease the polling frequency enough to miss the logs (?)

Reply:
https://github.com/iterative/cml/issues/413#issuecomment-1063415985

Reply:
Documented on https://github.com/iterative/cml/issues/413#issuecomment-1063415985

Reply:
We just only need to wait one window